### ISSUE TITLE
AssistedInject crashes without message when using with class with generic parameter

### ISSUE THREAD

**Issue by lwasyl** [square/AssistedInject#116]
AssistedInject fails without explanation when I try to use it with various classes that use generics. The simplest case that fails for me with Dagger 2.24, AssistedInject 0.5.0 and Kotlin 1.3.50:

```kotlin
class AClass<A> @AssistedInject constructor(
    @Assisted dependency: String
) {

    @AssistedInject.Factory
    interface Factory {

        fun create(dependency: String): AClass<String>
    }
}
```

The only error that appears is the following:
```
error: cannot find symbol
@dagger.Module(includes = {AssistedInject_PresentationModule.class})
                           ^
  symbol: class AssistedInject_PresentationModule
FAILURE: Build failed with an exception.
```

This is just a simplified version of what I was trying to do initially, which also fails with no meaningful message:

```kotlin
abstract class Base<T>
class Wrapper<A, B>

class AClass<A, B> @AssistedInject constructor(
    @Assisted dependency: (A) -> B
) : Base<Wrapper<A, B>>() {

    @AssistedInject.Factory
    interface Factory {

        fun <A, B> create(dependency: (A) -> B): AClass<A, B>
    }
}
```

**Comment by JakeWharton** [comment]
I wrote this test which passes:
```kotlin
  @Test fun stuff() {
    val input = JavaFileObjects.forSourceString("test.Test", """
      package test;

      import com.squareup.inject.assisted.Assisted;
      import com.squareup.inject.assisted.AssistedInject;

      abstract class Base<T> {}
      class Wrapper<A, B> {}
      interface KFunction1<A, B> {}

      class Test<A, B> extends Base<Wrapper<A, B>> {
        @AssistedInject
        Test(@Assisted KFunction1<? extends A, ? super B> dependency) {}

        @AssistedInject.Factory
        interface Factory {
          <A, B> Test<A, B> create(KFunction1<? extends A, ? super B> dependency);
        }
      }
    """)

    val expected = JavaFileObjects.forSourceString("test.Test_AssistedFactory", """
      package test;

      import java.lang.Override;
      import $GENERATED_TYPE;
      import javax.inject.Inject;

      $GENERATED_ANNOTATION
      public final class Test_AssistedFactory implements Test.Factory {
        @Inject public Test_AssistedFactory() {
        }

        @Override public <A, B> Test<A, B> create(KFunction1<? extends A, ? super B> dependency) {
          return new Test<A, B>(dependency);
        }
      }
    """)

    assertAbout(javaSource())
        .that(input)
        .processedWith(AssistedInjectProcessor())
        .compilesWithoutError()
        .and()
        .generatesSources(expected)
  }
```
Can you find the Java stub that is generated for `AClass` in your `build/` directory and show it?

**Comment by lwasyl** [comment]
Here's stub file from `build/tmp/kapt3/stubs/debug/com...`. I only removed `@kotlin.Metadata` annotations

For the first sample:
```java
package com.example;

import java.lang.System;

public final class AClass<A extends java.lang.Object> {
    
    @com.squareup.inject.assisted.AssistedInject()
    public AClass(@org.jetbrains.annotations.NotNull()
    @com.squareup.inject.assisted.Assisted()
    java.lang.String dependency) {
        super();
    }
    
    @com.squareup.inject.assisted.AssistedInject.Factory()
    public static abstract interface Factory {
        
        @org.jetbrains.annotations.NotNull()
        public abstract com.example.AClass<java.lang.String> create(@org.jetbrains.annotations.NotNull()
        java.lang.String dependency);
    }
}
```

For the second one:
```java
// AClass
package com.example;

import java.lang.System;

public final class AClass<A extends java.lang.Object, B extends java.lang.Object> extends com.example.Base<com.example.Wrapper<A, B>> {
    
    @com.squareup.inject.assisted.AssistedInject()
    public AClass(@org.jetbrains.annotations.NotNull()
    @com.squareup.inject.assisted.Assisted()
    kotlin.jvm.functions.Function1<? super A, ? extends B> dependency) {
        super();
    }
    
    @com.squareup.inject.assisted.AssistedInject.Factory()
    public static abstract interface Factory {
        
        @org.jetbrains.annotations.NotNull()
        public abstract <A extends java.lang.Object, B extends java.lang.Object>com.example.AClass<A, B> create(@org.jetbrains.annotations.NotNull()
        kotlin.jvm.functions.Function1<? super A, ? extends B> dependency);
    }
}

// Base
package com.example;

import java.lang.System;

public abstract class Base<T extends java.lang.Object> {
    
    public Base() {
        super();
    }
}

// Wrapper
package com.example;

import java.lang.System;

public final class Wrapper<A extends java.lang.Object, B extends java.lang.Object> {
    
    public Wrapper() {
        super();
    }
}
```

Also maybe irrelevant, but all classes are in an android library module with only single `debug` variant

**Comment by JakeWharton** [comment]
Ah, okay I think I found a problem in the simplified example! Still not in the second one though...

**Comment by lwasyl** [comment]
```
▶ java -version                                                                    
java version "1.8.0_202"
Java(TM) SE Runtime Environment (build 1.8.0_202-b08)
Java HotSpot(TM) 64-Bit Server VM (build 25.202-b08, mixed mode)
```

I have both targets for all projects set to Java 8, but if I can somehow check that my configuration is correct I can do that.

I can only see that in your test case variance is different:
```KFunction1<? extends A, ? super B> dependency```
While in my stubs it's
```Function1<? super A, ? extends B>```

**Comment by JakeWharton** [comment]
Ah, yeah I just guessed. Will try with the real thing, although since there's one of each it shouldn't make much of a difference.

**Comment by JakeWharton** [comment]
Narrowed in on the cause. The fix isn't so trivial though, but I'm working on it.

**Comment by lwasyl** [comment]
Seems like this is closed by #125, but fyi the test you added in #123 wasn't un-ignored and still fails. Not sure if it's the test that's incorrect or this was closed by mistake
